### PR TITLE
One 57117 fastlane add tests

### PR DESCRIPTION
**Pull request by cynicer**
- Added RSpec tests for the helper functions including the client update.
- No RSpec tests for the main action, as I'm unsure how to implement this. The main functionality is tested in the easyTravel pipeline anyway.

### CODE REVIEW

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 1]
```diff
@@ -26,6 +29,7 @@ def self.get_dss_client(params)
         if not response.kind_of? Net::HTTPSuccess
           error_msg = "Couldn't update #{dtxDssClientBin} (invalid response: #{response.message} (#{response.code})) for URL: #{self.to_redacted_api_token_string(clientUri)})"
           self.check_fallback_or_raise(dtxDssClientPath, error_msg)
+          return dtxDssClientPath
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:32]: bug 1

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 2]
```diff
@@ -59,30 +63,31 @@ def self.get_dss_client(params)
            File.exists?(dtxDssClientPath) and 
            File.read(versionFilePath) == remoteClientUrl and
            File.size(dtxDssClientPath) > 0)
-          # extract and save client
           updatedClient = false
 
-          # prevents from creating a StringIO object instead of FileIO in URI.open if <10kB
-          OpenURI::Buffer.send :remove_const, 'StringMax' if OpenURI::Buffer.const_defined?('StringMax')
-          OpenURI::Buffer.const_set 'StringMax', 0
+          # extract and save client
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:68]: added check for empty archive retrieved

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 3]
```diff
@@ -59,30 +63,31 @@ def self.get_dss_client(params)
...
+          zipped_tmp = self.save_to_tempfile(remoteClientUrl)
+          if File.size(zipped_tmp) <= 0
+            error_msg = "Downloaded symbolication client archive is empty (0 bytes)."
+            self.check_fallback_or_raise(dtxDssClientPath, error_msg)
+            return dtxDssClientPath
+          end
 
           begin
-            URI.open(remoteClientUrl) do |zipped|
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:70]: moved file download to separate function for testability -> remove one indent

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 4]
```diff
@@ -91,16 +96,19 @@ def self.get_dss_client(params)
                  Zip::EntryNameError, 
                  Zip::EntrySizeError, 
                  Zip::GPFBit3Error, 
-                 Zip::InternalError 
+                 Zip::InternalError,
+                 Zip::CompressionMethodError 
             error_msg = "Could not update/extract #{dtxDssClientBin}, please try again."
             self.check_fallback_or_raise(dtxDssClientPath, error_msg)
+            return dtxDssClientPath
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:103]: return path for completeness as well (doesn't change the outcome)

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 5]
```diff
@@ -91,16 +96,19 @@ def self.get_dss_client(params)
...
           end
 
           if updatedClient
             UI.success "Successfully updated DTXDssClient."
           else
             error_msg = "#{dtxDssClientBin} not found in served archive, please try again."
             self.check_fallback_or_raise(dtxDssClientPath, error_msg)
+            return dtxDssClientPath
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:111]: return path for completeness as well (doesn't change the outcome)

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 6]
```diff
@@ -117,23 +125,36 @@ def self.get_server_base_url(params)
       private
       def self.check_fallback_or_raise(fallback_client, error)
         UI.important "If this error persists create an issue on our Github project (https://github.com/Dynatrace/fastlane-plugin-dynatrace/issues) or contact our support at https://www.dynatrace.com/support/contact-support/."
+        UI.important error
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:128]: print the error message in both cases and don't omnit on fallback

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 7]
```diff
@@ -117,23 +125,36 @@ def self.get_server_base_url(params)
...
           raise error
         end
       end
 
+      # assumes the token parameter is appended last (there is only one parameter anyway)
       def self.to_redacted_api_token_string(url)
         urlStr = url.to_s
         str = "Api-Token="
         idx = urlStr.index(str)
-        token_len = urlStr.length - idx + str.length + 1
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:133]: bug 2: censored more than required from the token

File: lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb [hunk 8]
```diff
@@ -117,23 +125,36 @@ def self.get_server_base_url(params)
...
+        token_len = urlStr.length - idx - str.length
         urlStr[idx + str.length..idx + str.length + token_len] = "-" * token_len
         return urlStr
       end
+
+      def self.save_to_tempfile(url)
```
Comment (cynicer) [inline lib/fastlane/plugin/dynatrace/helper/dynatrace_helper.rb:147]: can't mock this part properly

**Review by harp79 — APPROVED** [review]

File: spec/dynatrace_action_spec.rb [hunk 9]
```diff
@@ -1,9 +1,9 @@
-describe Fastlane::Actions::DynatraceAction do
+describe Fastlane::Actions::DynatraceProcessSymbolsAction do
```
Comment (Landsi) [inline spec/dynatrace_action_spec.rb:1]: remove if not needed?

**Review by Landsi — APPROVED** [review]

**Pull request merged**